static/js/teachers.ts

```diff
@@ -547,102 +548,129 @@ export function add_account_placeholder() {
   }
 }
 
-export function generate_passwords() {
-    if (!$('#passwords_toggle').is(":checked")) {
-        $('.passwords_input').val('');
-        $('.passwords_input').prop('disabled', false);
-        return;
+export function toggleAutoGeneratePasswords() {
+    if ($('#passwords_toggle').is(':checked')) {
+        $('#passwords_toggle_checked_text').show();
+        $('#passwords_toggle_unchecked_text').hide();
+        $('#usernames_title').show();
+        $('#passwords_title').hide();
+        $('#usernames_desc').show();
+        $('#passwords_desc').hide();
+    } else {
+        $('#passwords_toggle_checked_text').hide();
+        $('#passwords_toggle_unchecked_text').show();
+        $('#usernames_title').hide();
+        $('#passwords_title').show();
+        $('#usernames_desc').hide();
+        $('#passwords_desc').show();
     }
-    $('.account_row').each(function () {
-        if ($(this).is(':visible')) {
-            $(this).find(':input').each(function () {
-                if ($(this).attr('id') == "password") {
-                    const random_password = generateRandomString(6);
-                    $(this).val(random_password);
-                }
-            });
-        }
-    });
-    $('.passwords_input').prop('disabled', true);
 }
 
-export function append_classname() {
-    const classname = <string>$('#classes').val();
-    $('.usernames_input').each(function () {
-        $(this).val($(this).val() + "_" + classname);
-    });
+export function printAccounts() {
+    var table = document.getElementById("accounts_table");
+    let newWindow = window.open("")!;
+    const css = `
+    <style>
+      @media print {
+        #accounts_table {
+          margin-top: 50px;
+          border-collapse: collapse;
+        }
+        #accounts_table td, th {
+          padding-left: 10px;
+          padding-right: 10px;
+          padding-top: 5px;
+          padding-bottom: 5px;
+          font-size: 24px;
+          border: 1px solid gray;
+        }
+      }
+    </style>`;
+    newWindow.document.write(table!.outerHTML + css);
```

Comment:
I think here instead of a bang we can use optional chainning with the `?` operator.